tests/e2e/core-tests/specs/shopper/front-end-cart-redirection.test.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable jest/no-export, jest/no-disabled-tests, jest/expect-expect */
+/**
+ * Internal dependencies
+ */
+ const {
+	shopper,
+	merchant,
+	createSimpleProduct,
+	setCheckbox,
+	unsetCheckbox,
+	settingsPageSaveChanges,
+} = require( '@woocommerce/e2e-utils' );
+
+/**
+ * External dependencies
+ */
+ const {
+	it,
+	describe,
+	beforeAll,
+	afterAll,
+} = require( '@jest/globals' );
+
+const config = require( 'config' );
+const simpleProductName = config.get( 'products.simple.name' );
+
+const runCartRedirectionTest = () => {
+	describe('Cart > Redirect to cart from shop', () => {
+		let simplePostIdValue;
+		beforeAll(async () => {
+			await merchant.login();
+			simplePostIdValue = await createSimpleProduct();
+
+			// Set checkbox in settings to enable cart redirection
+			await merchant.openSettings('products');
+			await setCheckbox('#woocommerce_cart_redirect_after_add');
+			await settingsPageSaveChanges();
+
+			await merchant.logout();
+		});
+
+		it('can redirect user to cart from shop page', async () => {
+			await shopper.goToShop();
+
+			// Add to cart from shop page
+			const addToCartXPath = `//li[contains(@class, "type-product") and a/h2[contains(text(), "${ simpleProductName }")]]` +
```

Comment:
We don't need to do it on this PR, but this would be great to break out into a shopper flow in `shopper.js`.